### ISSUE TITLE
Deploy to Heroku button

### ISSUE THREAD

**Issue by rarkins** [singapore/renovate#43]
Heroku button: https://devcenter.heroku.com/articles/heroku-button
app.json: https://devcenter.heroku.com/articles/app-json-schema
Scheduler: https://devcenter.heroku.com/articles/scheduler
Custom clock processes: https://devcenter.heroku.com/articles/scheduled-jobs-custom-clock-processes#custom-clock-processes

**Comment by rarkins** [comment]
Heroku Scheduler tasks need manual configuration through the web UI, but custom clock processes seem like they can be configured automatically. Here's a good howto: http://blog.andyjiang.com/intermediate-cron-jobs-with-heroku/

**Comment by rarkins** [comment]
app.json can specify environment variables that are mandatory. This is appropriate for RENOVATE_TOKEN, and perhaps we should add RENOVATE_REPOS (mandatory) and RENOVATE_FILES (optional). Both can be comma-separated

**Comment by rarkins** [comment]
Can't automatically start workers with "Deploy to Heroku": http://stackoverflow.com/questions/28128018/can-i-set-the-number-of-worker-processes-in-app-json

**Comment by rarkins** [comment]
https://scheduler.heroku.com/dashboard

**Comment by rarkins** [comment]
Custom clock processes count as a dyno running the entire month??

**Comment by hbetts** [comment]
Based on what I've found in the Heroku documentation, it seems like a clock process is nothing more than a normal worker process, but only ever scaled to one process that never sleeps, and relies on the developer to write the necessary code to kick off actions on a particular schedule.

Therefore, yes, you would be paying for a continuously running dyno regardless of how often you ran `renovate`.

**Comment by rarkins** [comment]
@hbetts thanks - I concluded the same. Right now I run `renovate` personally using Heroku Scheduler, deployed using `git`.

I got a Heroku Button working long ago, but have not merged it. 

The main problem I found was that you cannot "auto-configure" the Scheduler - its only configuring is through Heroku's web UI. I don't like having a "one click deploy" that actually then requires a bunch more clicks and typing afterwards to get working - I assumed people might be misled.

The second problem I found was the lack of good workflow for updates. I couldn't find any smooth way to either notify users of updates, let alone apply the updates.

Until there are solutions to these, I prefer to recommend Heroku users to deploy using `git`, so at leas they know what they're getting into.
